src/core/scan.ts

```diff
@@ -164,9 +159,21 @@ async function scanDir(
   });
   return fileNames
     .map((fullPath) => {
+      const path = relative(join(dir, name), fullPath);
+      const suffixMatch = path.match(suffixRegex);
```

Comment:
It is tricky to use _same_ suffix prefix because it also scans for HTTP verbs (`.get`, `.post`, etc) which is obviously not relevant for plugins for example.

I believe might be better idea if we keep `scanServerRoutes` intact and implement it in `scanPlugins`/`scanMiddleware`/etc (instead of `scanFiles` like now) with a shared util that extracts suffix. First step, you can just add it for `scanPlugins` for example then we refactor out.

Reply:
At first I did think of creating a utility function, but considering I needed to apply it everywhere I decided to implement it in `scanDir`. The performance impact was minimal as well as being a grouped match I can simply not use it in the plugins/middleware/tasks scans. But indeed this could become harder to maintain, let me fix this up.

Also, the minimal test suggested did run correctly and failed in a typo I've made